packages/block-editor/src/components/block-list/use-block-props/use-zoom-out-mode-exit.js

```diff
@@ -16,14 +16,17 @@ import { unlock } from '../../../lock-unlock';
  * @param {string} clientId Block client ID.
  */
 export function useZoomOutModeExit( { editorMode } ) {
-	const { getSettings } = useSelect( blockEditorStore );
-	const { __unstableSetEditorMode } = unlock(
+	const { getSettings, isZoomOut } = unlock( useSelect( blockEditorStore ) );
+	const { __unstableSetEditorMode, resetZoomLevel } = unlock(
 		useDispatch( blockEditorStore )
 	);
 
 	return useRefEffect(
 		( node ) => {
-			if ( editorMode !== 'zoom-out' ) {
+			// In "compose" mode.
+			const composeMode = editorMode === 'zoom-out' && isZoomOut();
```

Comment:
What are we exiting, the view or the selection mode? I guess view, so it doesn't matter the mode, only if `isZoomOut`. Right?

Reply:
This is one where I didn't know the answer. I kept the current behaviour on trunk which is that double click is connected to the mode.

As you say, let's merge and iterate 👍